### DISCUSSION TITLE
Profiling cFS apps

### DISCUSSION THREAD

**Discussion by srini-cmu** [nasa/cFS#473] [Q&A]
Is there a way to profile cFS apps? Specifically, when cFS is run on Linux (our use-case), it runs as one process. We would like to have some visibility on the CPU and memory utilization of each of the apps. Is there any APIs in cFS to get these values or tools we can run externally to tap these values?

**Comment by skliper** [comment]
The only thing I'm aware of within cFS (above the OS/platform) is the performance monitor APIs that time tag start/stop times for activities.  OS/compiler/platform level utilities are likely your best bet for anything lower level.
